zombie-monitor/src/main/scala/com/broadinstitute/dsp/zombieMonitor/DeletedOrErroredNodepoolChecker.scala

```diff
@@ -19,35 +21,64 @@ object DeletedOrErroredNodepoolChecker {
   def impl[F[_]](
     dbReader: DbReader[F],
     deps: KubernetesClusterCheckerDeps[F]
-  )(implicit F: Concurrent[F], logger: Logger[F], ev: Ask[F, TraceId]): CheckRunner[F, NodepoolToScan] =
-    new CheckRunner[F, NodepoolToScan] {
+  )(implicit F: Concurrent[F], logger: Logger[F], ev: Ask[F, TraceId]): CheckRunner[F, Nodepool] =
+    new CheckRunner[F, Nodepool] {
       override def appName: String = zombieMonitor.appName
 
-      override def resourceToScan: Stream[F, NodepoolToScan] = dbReader.getk8sNodepoolsToDeleteCandidate
+      override def resourceToScan: Stream[F, Nodepool] = dbReader.getk8sNodepoolsToDeleteCandidate
 
       // the report file will container all zombied nodepool, and also error-ed nodepool
       override def configs = CheckRunnerConfigs(s"deleted-or-errored-nodepools", false)
 
       override def dependencies: CheckRunnerDeps[F] = deps.checkRunnerDeps
 
-      def checkResource(nodepoolToScan: NodepoolToScan, isDryRun: Boolean)(implicit
+      def checkResource(nodepool: Nodepool, isDryRun: Boolean)(implicit
         ev: Ask[F, TraceId]
-      ): F[Option[NodepoolToScan]] =
+      ): F[Option[Nodepool]] = {
+        val isZombie = nodepool.cloudContext match {
+          case CloudContext.Gcp(project) =>
+            checkGkeNodepool(
+              nodepool.nodepoolId,
+              NodepoolId(KubernetesClusterId(project, nodepool.location, nodepool.clusterName), nodepool.nodepoolName),
+              isDryRun
+            )
+          case CloudContext.Azure(cloudContext) => checkAksNodepool(nodepool.nodepoolId, cloudContext, isDryRun)
+        }
+        F.ifF(isZombie)(nodepool.some, none[Nodepool])
+      }
+
+      def checkGkeNodepool(id: Long, gkeNodepoolId: NodepoolId, isDryRun: Boolean)(implicit
+        ev: Ask[F, TraceId]
+      ): F[Boolean] =
+        for {
+          nodepoolOpt <- deps.gkeService.getNodepool(gkeNodepoolId)
+          deleted <-
+            nodepoolOpt match {
+              case None =>
+                logger.info(s"Going to mark GKE nodepool ${id} and apps as DELETED") >>
+                  (if (!isDryRun) dbReader.markNodepoolAndAppDeleted(id) else F.unit) >>
+                  F.pure(true)
+              case Some(nodepool) if nodepool.getStatus == com.google.container.v1.NodePool.Status.ERROR =>
```

Comment:
I know this is pre-existing logic, but why do we change the status to error here?

Reply:
Especially since we do not do it for Azure, should we be consistent across cloud platforms?

Reply:
Yeah I guess if the GCP nodepool is Error, we set the Leo status to Error. I honestly don't remember why…. maybe we wanted to surface the cloud error to users instead of having it disappear?

Azure is kind of "dumber" in this regard -- Leo does not manage clusters/nodepools so here we're just testing the cloud cluster for existence, not whether it errored. I think this is reasonable to start with, but maybe could get more sophisticated if we see Azure nodepools start erroring.